src/Diffing/Interpreter.hs

```diff
@@ -29,11 +30,17 @@ stripDiff :: Functor syntax
           -> Diff.Diff syntax ann1 ann2
 stripDiff = bimap snd snd
 
-class DiffTerms term diff | diff -> term, term -> diff where
+class HasDiffFor (term :: * -> *) where
+  type DiffFor term = (res :: * -> * -> *) | res -> term
```

Comment:
`TypeFamilyDependencies`… you truly do not mess around, @robrix ❤️

Can you indicate in a comment why this dependency is necessary? It’s not obvious from the context.